bgpd/bgp_io.c

```diff
@@ -447,9 +447,9 @@ static uint16_t bgp_read(struct peer *peer)
 	size_t readsize; // how many bytes we want to read
 	ssize_t nbytes;  // how many bytes we actually read
 	uint16_t status = 0;
-	static uint8_t ibw[BGP_MAX_PACKET_SIZE * BGP_READ_PACKET_MAX];
+	uint8_t ibw[peer->max_packet_size * BGP_READ_PACKET_MAX];
```

Comment:
Currently BGP_READ_PACKET_MAX is set to 10 and max size can be 65535. Total allocation would be 6.5MB on stack for one variable.  Is there a reason behind making all these variable from making static to stack variable.

Reply:
Because they're VLAs. Actually this one is a good point, we probably shouldn't put this on the stack, as convenient as it is.